src/Docker/Client/Api.hs

```diff
@@ -93,6 +94,11 @@ startContainer sopts cid = requestUnit POST $ StartContainerEndpoint sopts cid
 stopContainer :: forall m. Monad m => Timeout -> ContainerID -> DockerT m (Either DockerError ())
 stopContainer t cid = requestUnit POST $ StopContainerEndpoint t cid
 
+-- | Blocks until a container with the given 'ContainerID' stops,
+-- then returns the exit code
+waitContainer :: forall m. Monad m => ContainerID -> DockerT m (Either DockerError ())
```

Comment:
I agree that we should return the exit code. [Here](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/wait-a-container) are the relevant docs. 

Does the exit code range from 0 to 255 (inclusive)? Seems like this should return an `Int` or maybe an [ExitCode](http://hackage.haskell.org/package/base-4.9.0.0/docs/System-Exit.html#t:ExitCode).